ckanext/stadtzhtheme/plugin.py

```diff
@@ -746,8 +749,39 @@ def _set_resource_filename(self, resource):
             if upload.filename:
                 resource["filename"] = os.path.basename(upload.filename)
 
+    def _set_markdown_snippet_text(self, resource):
+        if not resource.get("markdown_snippet"):
```

Comment:
Could we check if the resource *has* the snippet and return if so, and then do the rest of this code?